shared/common-adapters/tab-bar.native.js

```diff
@@ -34,15 +34,24 @@ class TabBarButton extends Component<void, TabBarButtonProps, void> {
     const backgroundColor = this.props.selected ? globalColors.darkBlue4 : globalColors.midnightBlue
     const badgeNumber = this.props.badgeNumber || 0
 
+    let badgeComponent
+    if (this.props.badgePosition === 'top-right') {
```

Comment:
What I am saying is that the tab-bar doesnt' need to understand some idea of 'badgePosition' it can just plumb through an additional prop called 'badgeStyle' which essentially allows any user of the tab-bar to specify their own offets. This hardcoded idea of being -40 and -20 won't work anywhere else and isn't scalable.  you can just do ```badgeStyle={{marginLeft: 5, ...this.props.badgeStyle}}``` and set badgeStyle in the mobile nav and you're done